File: Native/Android/MoEPLPnIntegration/app/src/main/java/com/vamsi/moe_pl_pn_integration/push/PlotlinePushProvider.kt
package com.vamsi.moe_pl_pn_integration.push

import android.content.Context
import android.util.Log
import com.google.firebase.messaging.RemoteMessage
import so.plotline.insights.PlotlinePush

/**
 * Plotline push provider.
 *
 * Detection and rendering delegate to Plotline's own APIs. The Plotline SDK
 * merges no receiver/service, so it depends entirely on the host app feeding
 * payloads and the FCM token to it.
 */
class PlotlinePushProvider : PushProvider {

    override val name: String = "Plotline"

    override fun canHandle(data: Map<String, String>): Boolean =
        PlotlinePush.isPushPlotline(data)

    override fun handle(context: Context, message: RemoteMessage) {
        PlotlinePush.showNotification(context, message.data)
    }

    override fun onNewToken(context: Context, token: String) {
        Log.d("Token Called" + "PL", token)
        PlotlinePush.setFcmToken(context, token)
    }
}

File: Native/iOS/CT-PL-Pn-Integration/CT-PL-Pn-Integration/Push/PlotlinePushProvider.swift
import Foundation
import UserNotifications
import Plotline

/// Plotline push provider.
///
/// Detection and rendering delegate to Plotline's own APIs. The Plotline SDK
/// merges no receiver/service, so it depends entirely on the host app feeding
/// payloads and the APNs token to it.
final class PlotlinePushProvider: PushProvider {

    var name: String { "Plotline" }

    func canHandle(request: UNNotificationRequest) -> Bool {
        PlotlinePush.isPushPlotline(request: request)
    }

    func handleWillPresent(_ notification: UNNotification,
                           completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
        PlotlinePush.onNotificationReceived(notification: notification,
                                            completionHandler: completionHandler)
    }

    func handleNotificationTap(_ response: UNNotificationResponse,
                               completionHandler: @escaping () -> Void) {
        PlotlinePush.userNotificationCenter(center: UNUserNotificationCenter.current(),
                                            didReceive: response)
        completionHandler()
    }

    func onNewToken(_ deviceToken: Data) {
        PlotlinePush.setPushToken(deviceToken: deviceToken)
    }
}